### PR TITLE
parsers: add nim parser

### PR DESCRIPTION
**Pull request by theHamsta**
~~WIP: not feasible at the moment due to SEGFAULT during editor session https://github.com/helix-editor/helix/issues/3117#issuecomment-1219922184~~ SEGFAULT has been fixed :tada: 

- [x] TODO: try out Nim  https://github.com/haxscramper/tree-sitter-nim -> I like https://github.com/aMOPel/tree-sitter-nim better

Parser is not indented for final use yet. PR for anyone who wants to follow the progress of the parser.

### CODE REVIEW

File: lua/nvim-treesitter/parsers.lua [hunk 1]
```diff
@@ -1166,6 +1166,15 @@ list.gitignore = {
   maintainers = { "@theHamsta" },
 }
 
+list.nim = {
+  install_info = {
+    url = "https://github.com/aMOPel/tree-sitter-nim",
```
Comment (theHamsta) [inline lua/nvim-treesitter/parsers.lua:1171]: you can change this path to your local development directory

File: queries/nim/highlights.scm [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  "="
+] @operator
+
+(importStmt (keyw) @include)
+(fromStmt (keyw) @include)
+(str_lit) @string
+(int_lit) @number
+[(float_lit) (float_suffix)] @float
+
+(comment) @comment
```
Comment (sigmaSd) [inline queries/nim/highlights.scm:27]: I think you should also add
`(docComment) @comment`

Comment (sigmaSd) [reply queries/nim/highlights.scm:27]: Actually maybe all of these
image

Comment (theHamsta) [reply queries/nim/highlights.scm]: Thank you! I integrated your suggestion!

**Comment by sigmaSd** [comment]
typeDesc no longer works, it seems related to this commit https://github.com/aMOPel/tree-sitter-nim/commit/795aef0df5385f36283a56512243e2e368db959d

**Comment by sigmaSd** [comment]
I put this branch in a plugin, so at least we can have some syntax highlighting easily for nim https://github.com/sigmaSd/nim-nvim-basic

**Comment by j-james** [comment]
Hi, see also https://github.com/helix-editor/helix/pull/6123. The queries need to be tested, but once they work I was planning on PR'ing them into aMOPel's repository.

**Pull request closed without merging**